website/docs/advanced-concepts/access-control/granular-access-control/roles.md

```diff
@@ -0,0 +1,242 @@
+| **Export**         |Grants users permission to export applications in a workspace. | [App Resources](#application-resources-permissions) | 
+| **Invite**         |Grants users permission to invite users to a user group | [Groups and roles](#groups-and-roles-permissions)     |
+| **Make public**    |Grants users permission to make the applications public in a workspace |[App Resources](#application-resources-permissions)|
+| **Execute**        |Grants users permission to execute queries on a datasource | [Datasource and queries](#datasource-and-queries-permissions) |
+| **Remove users**   |Grants users permission to remove a user from a group| [Groups and roles](#groups-and-roles-permissions) |
+
+Let's understand these permissions in detail.
+
+### Application resources permissions
+
+On an application level, here are the permissions that can be assigned to user or a group -
+
+#### Create
+
+With this permission, users can create applications, pages, API queries and JS Objects. This also automatically enables permission to view, edit, and delete existing pages, API, queries and JS Objects. Under Application Resources, you can control the **Create** permission at different hierarchical levels:
+
+##### Workspace 
+
+When you grant the **Create** permission to a workspace, it provides access to create applications only in that particular workspace. To enable permission to create workspaces, go to the [Others](#others-permissions) tab.
+
+
+##### Application
+
+When you grant the **Create** permission to an application, it provides access to create new pages in the respective application.
+
+##### Page
+
+At a more granular level, giving **Create** permission to a specific page allows the user to create API, queries and JS Objects for the respective page. 
+
+#### Edit
+
+**Edit** allows the user to modify the existing applications, pages, API, queries and JS Objects in a workspace. You can edit the app name, pages, and their order, color, icon, theme etc. Edit also enables access to deploy an app and connect to a Git repository. 
+Edit permission can be given at different hierarchical levels -
+
+##### Workspace 
+
+When you grant the **Create** permission to a workspace, it provides access to edit the newly created applications in that particular workspace. 
+
+##### Application
+
+If you want the user to have edit access only to a specific application in a workspace, open the respective workspace tree and check the edit permission adjacent to the respective application. **Edit** for just an application allows you to edit the pages of that application.
+
+##### Page
+
+At a more granular level, giving **Edit** permission to a specific page allows the user to edit API, queries and JS Objects of the respective page. You can edit the page name, clone the page, set it as the home page, and toggle their visibility.
+
```

Comment:
Edit is also present for Queries/APIs/JS objects. Should we mention the functionality for them as well?

Reply:
It is under the page permissions right? I think this should be enough.